common/functions/src/scalars/semi_structureds/parse_json.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::sync::Arc;
+
+use common_datavalues::prelude::*;
+use common_datavalues::with_match_physical_primitive_type;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use serde_json::Value as JsonValue;
+
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+use crate::scalars::FunctionDescription;
+
+pub type TryParseJsonFunction = ParseJsonFunctionImpl<true>;
+
+pub type ParseJsonFunction = ParseJsonFunctionImpl<false>;
+
+#[derive(Clone)]
+pub struct ParseJsonFunctionImpl<const SUPPRESS_PARSE_ERROR: bool> {
+    display_name: String,
+}
+
+impl<const SUPPRESS_PARSE_ERROR: bool> ParseJsonFunctionImpl<SUPPRESS_PARSE_ERROR> {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(ParseJsonFunctionImpl::<SUPPRESS_PARSE_ERROR> {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        let features = FunctionFeatures::default().deterministic().num_arguments(1);
+        FunctionDescription::creator(Box::new(Self::try_create)).features(features)
+    }
+}
+
+impl<const SUPPRESS_PARSE_ERROR: bool> Function for ParseJsonFunctionImpl<SUPPRESS_PARSE_ERROR> {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        if args[0].data_type_id() == TypeID::Null {
+            return Ok(NullType::arc());
+        }
+
+        Ok(Arc::new(NullableType::create(VariantType::arc())))
+    }
+
+    fn eval(&self, columns: &ColumnsWithField, input_rows: usize) -> Result<ColumnRef> {
+        let column = columns[0].column();
+        if column.data_type_id() == TypeID::Null {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
```

Comment:
Since we use `passthrough_null`, no need to check TypeID::Null anymore.

Reply:
ok